specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/apimanagement.json

```diff
@@ -0,0 +1,867 @@
+      "x-ms-enum": {
+        "name": "NotificationName",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "RequestPublisherNotificationMessage",
+            "description": "The following email recipients and users will receive email notifications about subscription requests for API products requiring approval."
+          },
+          {
+            "value": "PurchasePublisherNotificationMessage",
+            "description": "The following email recipients and users will receive email notifications about new API product subscriptions."
+          },
+          {
+            "value": "NewApplicationNotificationMessage",
+            "description": "The following email recipients and users will receive email notifications when new applications are submitted to the application gallery."
+          },
+          {
+            "value": "BCC",
+            "description": "The following recipients will receive blind carbon copies of all emails sent to developers."
+          },
+          {
+            "value": "NewIssuePublisherNotificationMessage",
+            "description": "The following email recipients and users will receive email notifications when a new issue or comment is submitted on the developer portal."
+          },
+          {
+            "value": "AccountClosedPublisher",
+            "description": "The following email recipients and users will receive email notifications when developer closes his account."
+          },
+          {
+            "value": "QuotaLimitApproachingPublisherNotificationMessage",
+            "description": "The following email recipients and users will receive email notifications when subscription usage gets close to usage quota."
+          }
+        ]
+      },
+      "description": "Notification Name Identifier.",
+      "x-ms-parameter-location": "method"
+    },
+    "NotifySubscriptionStateChangeParameter": {
+      "name": "notify",
+      "in": "query",
+      "required": false,
+      "type": "boolean",
+      "description": "Notify change in Subscription State. \n - If false, do not send any email notification for change of state of subscription \n - If true, send email notification of change of state of subscription ",
+      "x-ms-parameter-location": "method"
+    },
+    "OpenIdConnectIdParameter": {
+      "name": "opid",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Identifier of the OpenID Connect Provider.",
+      "maxLength": 256,
+      "pattern": "^[^*#&+:<>?]+$",
+      "x-ms-parameter-location": "method"
+    },
+    "OperationIdParameter": {
+      "name": "operationId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Operation identifier within an API. Must be unique in the current API Management service instance.",
+      "minLength": 1,
+      "maxLength": 80,
+      "x-ms-parameter-location": "method"
+    },
+    "ResolverIdParameter": {
```

Comment:
>ResolverIdParameter [](http://example.com/codeflow?start=5&length=19)

Why is this added here ? Does not look like this is used anywhere in this file?

Reply:
should this be in the definitions file instead ?

Reply:
I am not sure about the origins of the pattern but our team seems to add parameters to the apimanagement.json file and refer to them in the apimapis.json. I can try to dig and try to find out why if moving these to the definitions files is preferred.